src/engines/TimeToLive.js

```diff
@@ -0,0 +1,149 @@
+import DefaultHashTable from '../dataStructure/HashTable';
+import DoublyLinkedList from '../dataStructure/DoublyLinkedList';
+import * as hashTableProp from '../hashTableSymbol';
+
+function TimeToLive({ HashTable = DefaultHashTable, defaultTTL } = {}) {
+  const store = new HashTable();
+  const timePartition = new HashTable();
+  const timeIndexInterval = 5 * 60 * 1000; // milliseconds.
+
+  let lowestTimePartition = Date.now();
+
+  this.add = (key, value, ttl = defaultTTL) => {
+    if (!ttl || !Number.isInteger(ttl) || ttl <= 0)
+      throw Error(
+        'Expected ttl value (should be positive integer). ' +
+          'you can have to mention it in add method or mention as defaultTTL at constructor',
+      );
+
+    const expireTTL = Date.now() + ttl;
+    const bucket = getTimeBucket(expireTTL);
+    bucket.addFirst(key);
+    const tNode = bucket.getFirstNode();
+    const payload = { value, ttl: expireTTL, tNode };
```

Comment:
ttl is misleading here. Per definition https://en.wikipedia.org/wiki/Time_to_live is a timespan. expireTTL is a timestamp. So ```expireTime``` or short ```expires``` is more precise,